src/components/URLImportDialog/index.js

```diff
@@ -0,0 +1,255 @@
+import React, { useState } from "react";
+
+import { makeStyles } from "@material-ui/core/styles";
+import { useTheme } from "@material-ui/core/styles";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+
+import UploadIcon from "mdi-react/UploadIcon";
+import Button from "@material-ui/core/Button";
+import CloseIcon from "@material-ui/icons/Close";
+import Dialog from "@material-ui/core/Dialog";
+import DialogActions from "@material-ui/core/DialogActions";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import IconButton from "@material-ui/core/IconButton";
+import TextField from "@material-ui/core/TextField";
+
+import intlData from "./messages";
+import ids from "./ids";
+
+import {
+    build as buildID,
+    formatMessage as fmt,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+
+import { injectIntl } from "react-intl";
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        flexGrow: 1,
+    },
+    close: {
+        color: theme.palette.blueGrey,
+    },
+    container: {
+        display: "flex",
+        alignItems: "center", // without this the import button balloons in size vertically.
+
+        // Align things vertically on mobile.
+        [theme.breakpoints.down("sm")]: {
+            justifyContent: "center",
+            flexDirection: "column",
+        },
+    },
+    urlField: {
+        width: "100%",
+        margin: "0 10px 0 0",
+
+        [theme.breakpoints.down("sm")]: {
+            margin: 0, // The textbox shifts to the left without this.
+        },
+    },
+    closeDialog: {
+        position: "absolute",
+        right: theme.spacing(1),
+        top: theme.spacing(1),
+        color: theme.palette.blueGrey,
+    },
+    importDefault: {
+        background: theme.palette.primary.main,
+        marginBottom: 25,
```

Comment:
I could be wrong, but I think the `theme.spacing(#)` stuff can be used for the margins in your styles

Reply:
I can use it in most places, so I switched it over. There was one spot where things broke horribly, so I changed it back.